src/walk_dirs.rs

```diff
@@ -144,13 +155,120 @@ pub fn discover_fences_and_files(start_path: &str) -> Vec<WalkFileData> {
     .collect();
 }
 
+
+fn get_imports_map(imports: &Vec<SourceSpecifiers>, file_path: &PathBuf) -> HashMap<String, Option<HashSet<String>>> {
+    let mut imports_map : HashMap<String, Option<HashSet<String>>> = HashMap::new();
+    // imports.iter().for_each(|import| {
+    //   import.specifiers
+    // });
+    imports.iter().for_each(|import| {
+    
+      let mut set: HashSet<String> = import.specifiers.iter().filter_map(|spec| -> Option<String> {
+        let file_text = std::fs::read(file_path).expect(&format!(
+          "error opening source file \"{:?}\"",
+          file_path
+        ));
+        if let Some(default) = spec.as_default() {
+          let text =  &file_text[default.span.lo().to_usize()-1..default.span.hi().to_usize()-1];
+          return Some(String::from_utf8_lossy(text).to_string());
+        }
+        if let Some(named) = spec.as_named() {
+          
+          let text =  &file_text[named.span.lo().to_usize()-1..named.span.hi().to_usize()-1];
+          println!("{}", String::from_utf8_lossy(text));
+          return Some(String::from_utf8_lossy(text).to_string());
+        }
+        None
+      }).collect();
+
+      if let Some(current_set) = imports_map.get(&import.source.value.to_string()) {
+        if let  Some(current_set) = current_set {
+          let mut new_set: HashSet<String> = HashSet::from_iter(current_set.iter().map(|v| v).cloned());
+          for val in set {
+            new_set.insert(val);
+          }
+          imports_map.insert(import.source.value.to_string(), Some(new_set.to_owned()));
+        } else {
+
+        }
+      } else {
+        if set.is_empty() {
+          imports_map.insert(import.source.value.to_string(), None);
+        } else {
+          imports_map.insert(import.source.value.to_string(), Some(set));
+        }
+      }
+    
+    });
+    imports_map
+}
+
+pub fn create_lexer<'a>(fm: &'a swc_common::SourceFile) -> Lexer<'a, StringInput<'a>> {
+    let lexer = Lexer::new(
+      Syntax::Typescript(Default::default()),
+      Default::default(),
+      StringInput::from(fm),
+      None
+    );
+    lexer
+}
+
+pub struct SourceSpecifiers {
+  specifiers: Vec<swc_ecma_ast::ImportSpecifier>,
+  source: Str
+}
+
+pub fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Vec<SourceSpecifiers>{
+  let cm = Arc::<swc_common::SourceMap>::default();
+  let fm = cm.load_file(Path::new(file_path.to_str().unwrap())).expect("Could not load file");
+  let handler = Handler::with_tty_emitter(ColorConfig::Auto, true, false, Some(cm.clone()));
+  
+  let lexer = create_lexer(&fm);
+
+  let capturing = Capturing::new(lexer);
+
+  let mut parser = Parser::new_from(capturing);
+
+  for e in parser.take_errors() {
+    e.into_diagnostic(&handler).emit();
+  }
+
+
+
+  let ts_module = parser.parse_typescript_module()
+    .map_err(|e| e.into_diagnostic(&handler).emit())
+    .expect("Failed to parse module.");
```

Comment:
You should handle these errors and return Result<> here, since the program shouldn't _crash_ when it encounters invalid input

Reply:
Specifically, it's expected for devs to run it against unparseable code and we should give them a human readable error (and a partial fence evaluation) rather than crashing with a stack trace when that happens